### ISSUE TITLE
rate limiting per client IP

### ISSUE THREAD

**Issue by 0xERR0R** [0xERR0R/blocky#1135]
It should be possible to configure rate limit per client

**Comment by ThinkChaos** [comment]
I'm not against adding this, but I think we should at make it clear in the docs this isn't enough if you want to prevent (D)DOS and is more for broken clients/someone testing too aggressively.  
That being said, IMO something like [fail2ban](https://github.com/fail2ban/fail2ban)/[crowdsec](https://github.com/crowdsecurity/crowdsec) would be a better tool to achieve this since it works with a firewall and will avoid the cost of blocky processing more requests by that client just to ignore them, and can block all traffic from the offenders.

**Comment by 0xERR0R** [comment]
Yes, absolutely agree. I think, we should provide a basic layer of security out of the box. This is the same thing as SSL certificate. It is better to configure SSL termination in a reverse proxy and put blocky behind it without any encryption. But for users who just want to run blocky on a small edge device it is too much overhead.

**Comment by h3ndrik** [comment]
Hey, can we get this? Or at least some documentation on how to configure fail2ban or something? Today I found out I was sending out 8MB of traffic constantly and involuntarily participating in some DNS amplification attacks.
